### ISSUE TITLE
prefer_iterable_whereType false positive

### ISSUE THREAD

**Issue by gabrc** [dart-lang/linter#1520]
The code below is flagged by the lint rule prefer_iterable_whereType:
```
var foo = [42].where((num) => num is! int);
```

**Comment by pq** [comment]
Great catch.  Thanks for the report @gabrc !
